how-to/misc/multiregion-deployments.rst

```diff
@@ -0,0 +1,154 @@
+Multi-region deployments
+========================
+
+OpenStack deployments can span across multiple regions while sharing only the
+identity (Keystone) and dashboard (Horizon) services.
+
+Commonly used with large deployments, the OpenStack region concept is
+quite flexible. These may be geographically distinct locations, providing
+regional level fault tolerance.
+
+Alternatively, regions can exist within the same physical site, used to
+partition a large cluster and distribute load among critical components like
+databases or message brokers.
+
+Manual mode
+-----------
+
+Region controllers
+~~~~~~~~~~~~~~~~~~
+
+To accommodate multi-region environments, Canonical OpenStack defines the
+region controller role, which is mutually exclusive with all other roles.
+
+Region controllers will only run the shared services, such as Keystone and
+Horizon.
+
+In local mode, a region controller may be bootstrapped like so:
+
+::
+
+	sunbeam cluster bootstrap --role region_controller
+
+For high availability, additional region controllers can be added using
+the standard procedure. First, obtain a cluster join token:
+
+::
+
+	sunbeam cluster add $fqdn
+
+Then use the token to join the new region controller node:
+
+::
+
+	sunbeam cluster join --role region_controller $token
+
+Note that standard Sunbeam deployments can also act as primary regions.
+
+Secondary regions
+~~~~~~~~~~~~~~~~~
+
+The secondary regions are distinct Canonical Openstack deployments, using
+separate Kubernetes clusters and Juju controllers.
+
+Shared services such as Keystone and Horizon will be omitted from secondary
+regions. These services will be provided by the region controllers and
+consumed through cross-controller Juju relations.
+
+In order to access the region controller, each secondary region node will
+need a token obtained from the region controllers:
+
+::
+
+	sunbeam cluster add-secondary-region-node $fqdn
+
+The token must then be passed to the bootstrap command:
+
+::
+
+	sunbeam cluster bootstrap \
+    	--role control,compute,storage --region-controller-token=$token
+
+During bootstrap, make sure to specify a region name other than the one of the
+region controller.
+
+A region controller token is also required when joining secondary region nodes.
+This means that the join operation requires two tokens: one from the region
+controller and one from an existing member of the secondary region.
+
+::
+
+	sunbeam --verbose cluster join --role control,compute,storage \
+    	--region-controller-token $region_ctrl_token $same_region_token
```

Comment:
I'm not really sure why, this UX seems a bit broken? is that to be able to create necessary juju credentials?

Reply:
I discussed this with Hemanth a month ago and this was the agreed approach. Since we don't want to share credentials for security reasons, we need a registration token from the region controller.